planning/db/taxprofiler_parser.py

```diff
@@ -0,0 +1,177 @@
+import re
+import os
+import json
+import argparse  # Import argparse module for command-line argument parsing
+
+# Function to parse samtools stats files and extract QC metrics
+def parse_samtools_stats(file_path):
+    with open(file_path, 'r') as f:
+        stats_data = f.read()
+    
+    total_bases = int(re.search(r'SN\ttotal length:\s+(\d+)', stats_data).group(1))
+    raw_reads = int(re.search(r'SN\traw total sequences:\s+(\d+)', stats_data).group(1))
+    reads_mapped = int(re.search(r'SN\treads mapped:\s+(\d+)', stats_data).group(1))
+    reads_unmapped = int(re.search(r'SN\treads unmapped:\s+(\d+)', stats_data).group(1))
+    trimmed_reads = int(re.search(r'SN\tbases trimmed:\s+(\d+)', stats_data).group(1))
```

Comment:
If `re.search(...)` returns `None` instead of a `Match[str]`, then this part will crash since `.group(1)` is a method of `Match` and not `None`. I think it's fine for now, given the time and state of the project, but I think it's a good idea that this is resolved soon :)